app/controllers/check_ins_controller.rb

```diff
@@ -0,0 +1,31 @@
+class CheckInsController < ApplicationController
+  # before_action :setup_variables, only: [:new, :create]
+  def index
+    @check_in = CheckInDecorator.new(check_in_params)
+  end
+
+  def create
+    if @check_in.save do |check_in_creation|
```

Comment:
P.s. the notion of "What variables and methods are available to which methods?" is known as *scoping*. It's one of the things that is super easy to get wrong, trying to access a method or variable that doesn't exist in *this* method because it looks like a variable that's available from another method.